driver/sql/projection_notification_processor_test.go

```diff
@@ -0,0 +1,106 @@
+package sql_test
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/hellofresh/goengine/driver/sql"
+	mocks "github.com/hellofresh/goengine/mocks/driver/sql"
+	"github.com/stretchr/testify/require"
+)
+
+func TestStartProcessor(t *testing.T) {
+	testCases := []struct {
+		title        string
+		notification func() *sql.ProjectionNotification
+		queueFunc    string
+	}{
+		{
+			"Handle nil notification",
+			func() *sql.ProjectionNotification {
+				return nil
+			},
+			"Queue",
+		},
+		{
+			"Handle new notification",
+			func() *sql.ProjectionNotification {
+				return &sql.ProjectionNotification{
+					No:          1,
+					AggregateID: "abc",
+				}
+			},
+			"ReQueue",
+		},
+		{
+			"Handle retried notification",
+			func() *sql.ProjectionNotification {
+				return &sql.ProjectionNotification{
+					No:          1,
+					AggregateID: "abc",
+					ValidAfter:  time.Now().Add(time.Millisecond * 200),
+				}
+			},
+			"ReQueue",
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.title, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+
+			nqMock := mocks.NewMockNotificationQueueInterface(ctrl)
+			ctx := context.Background()
+			notification := testCase.notification()
+
+			e := nqMock.EXPECT()
+			queueCallCount := 0
+			reQueueCallCount := 0
+			switch testCase.queueFunc {
```

Comment:
Seems to me like you are testing multiple flows with the same test func. I'm fine with this but maybe next time split it into 2 different tests?